ext/src/ruby_api/wasi_context.rs

```diff
@@ -0,0 +1,56 @@
+use super::{root, WasiCtxBuilder, wasi_ctx_builder::{file_r, file_w, wasi_file}};
+use crate::error;
+use std::{fs::File, path::PathBuf, cell::RefCell};
+use magnus::{class, method, function, prelude::*, RTypedData, Error, Ruby, Value, RString};
+use wasmtime_wasi::WasiCtx as WasiCtxImpl;
+use deterministic_wasi_ctx::build_wasi_ctx as wasi_deterministic_ctx;
+use wasi_common::pipe::ReadPipe;
+
+#[magnus::wrap(class = "Wasmtime::WasiContext")]
```

Comment:
I think we should also consider adding `#[magnus::wrap(class = "Wasmtime::WasiContext", size, free_immediately)]`; `size` and `free_immediately`

Ref: https://docs.rs/magnus/latest/magnus/attr.wrap.html

Reply:
I also wonder if this should be exposed as `WasiCtx` to Ruby to reflect the actual naming in the Rust API. 

https://docs.wasmtime.dev/api/wasmtime_wasi/struct.WasiCtx.html